src/transformers/modeling_utils.py

```diff
@@ -2109,7 +2098,26 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                     "Using `low_cpu_mem_usage=True` or a `device_map` requires Accelerate: `pip install accelerate`"
                 )
 
+        if load_in_8bit and bitsandbytes_config is not None:
+            raise ValueError(
+                "You can't pass both `load_in_8bit=True` and `bitsandbytes_config` as they are mutually exclusive."
+            )
+        elif bitsandbytes_config is None and load_in_8bit:
+            # Use the default config if `load_in_8bit` is True.
+            from .utils.bitsandbytes_config import BitsandbytesConfig
+
+            bitsandbytes_config = BitsandbytesConfig()
+        elif bitsandbytes_config is not None:
+            # We force `load_in_8bit` to True if `bitsandbytes_config` is passed.
+            load_in_8bit = True
+
         if load_in_8bit:
+            # warn that this argument will be deprecated in the future and fully replaced with `bitsandbytes_config`
+            logger.warning(
+                "The `load_in_8bit` argument will be deprecated in the future (v5). Please use "
+                "`BitsandbytesConfig` instead."
+            )
```

Comment:
Ah, so that's your plan. I actually like having a simple `load_in_8bit=True` that loads the most basic config.